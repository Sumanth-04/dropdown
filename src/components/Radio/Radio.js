import React, { Component } from "react";
import "./Radio.scss";

class Radio extends Component {
  render() {
    const { label, children, className } = this.props;
    return (
      <span className="containers">
        <input
          type="radio"
          className = 'rd-radio'
          {...this.props}
        /><span class="checkmark"></span>
        {" "}
        {label ? label : children}
      </span>
    );
  }
}

export default Radio;
