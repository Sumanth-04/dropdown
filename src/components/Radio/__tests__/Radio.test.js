import React from "react";
import renderer from "react-test-renderer";
import Radio from "../Radio";

it("Radio should have a child or label", () => {
  const tree = renderer
    .create(<Radio className="Radio" name="radAnswer" label="click" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
