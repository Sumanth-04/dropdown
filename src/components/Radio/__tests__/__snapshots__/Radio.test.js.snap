// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should have a child or label 1`] = `
<span
  className="containers"
>
  <input
    className="Radio"
    label="click"
    name="radAnswer"
    type="radio"
  />
  <span
    class="checkmark"
  />
   
  click
</span>
`;
